### PR TITLE
Bugfix/poor error on missing latexml

### PR DESCRIPTION
**Pull request by CAMOBAP795**
- #96 
 - #101 

I missed the point why https://github.com/metanorma/metanorma-standoc/pull/100 was merged, maybe some serious issue was introduced by https://github.com/metanorma/metanorma-standoc/pull/98, please let me know

I did an ugly fix for windows tests failing, to provide better fix I need look at plantuml in the deep, or add some file check with timeout

### CODE REVIEW

**Comment by ronaldtse** [comment]
@CAMOBAP795 thanks for this, but both Travis and AppVeyor are failing for now :wink:

**Comment by CAMOBAP795** [comment]
@ronaldtse, yes) but now failure tests are the same for both windows and *nix 

I think it's related to xml changes (which being downloaded)

**Comment by CAMOBAP795** [comment]
*I will try to fix it today

**Comment by CAMOBAP795** [comment]
@ronaldtse @opoudjis I don't see other options than just update unit tests logic again, is that good solution?

**Comment by ronaldtse** [comment]
@CAMOBAP795 I'm completely fine with updating the unit tests to be correct. However, what exactly is the difference right now?

Have you done a diff (without regard to whitespace)?

File: lib/asciidoctor/standoc/macros.rb [hunk 1]
```diff
@@ -88,6 +88,7 @@ def self.generate_file parent, reader
         fn = save_plantuml parent, reader, localdir
         umlfile = Pathname.new(localdir) + "plantuml" + "#{fn}.pml"
         system "plantuml #{umlfile}"
+        sleep(2) if Gem.win_platform?
```
Comment (ronaldtse) [inline lib/asciidoctor/standoc/macros.rb]: @CAMOBAP795 why do we need this strange line here?

File: lib/metanorma/standoc/latexml_requirement.rb [hunk 2]
```diff
@@ -8,6 +8,8 @@ def initialize
         @minimal_version = '0.8.0'
         version_output, = Open3.capture2e("latexml --VERSION")
         @actual_version = version_output&.match(%r{\d+(.\d+)*})
+      rescue
+        warn "LaTeXML not found in PATH, please make sure that you installed LaTeXML"
```
Comment (ronaldtse) [inline lib/metanorma/standoc/latexml_requirement.rb]: "LaTeXML not found in `PATH`, please ensure that LaTeXML is installed."

File: lib/metanorma/standoc/processor.rb [hunk 3]
```diff
@@ -8,8 +8,6 @@ def initialize
         @short = :standoc
         @input_format = :asciidoc
         @asciidoctor_backend = :standoc
-
```
Comment (ronaldtse) [inline lib/metanorma/standoc/processor.rb]: Only when a `[latexmath]` block is encountered we need LaTeXML. @opoudjis any tips here?

File: lib/asciidoctor/standoc/macros.rb (hunk 1 above)
Comment (CAMOBAP795) [reply lib/asciidoctor/standoc/macros.rb]: This is a workaround for windows issue https://ci.appveyor.com/project/ribose/metanorma-standoc/builds/26662260/job/b6ro2ct4tjh242g3 (ugly workaround but I didn't find anything better for now)

I think under the hood in plantuml something wrong with waiting for process https://github.com/plantuml/plantuml/blob/2c655a6b7481c39a41fc0dc57b561bd1db1ecfe2/src/net/sourceforge/plantuml/cucadiagram/dot/AbstractGraphviz.java#L109

I propose to apply this workaround and fix it later in the scope of another ticket

File: lib/metanorma/standoc/latexml_requirement.rb (hunk 2 above)
Comment (CAMOBAP795) [reply lib/metanorma/standoc/latexml_requirement.rb]: Fixed

**Comment by CAMOBAP795** [comment]
> However, what exactly is the difference right now?

5 tests failed, I suppose this is because docs downloaded from https://www.iso.org

@opoudjis @ronaldtse is that ok that we do whole doc validation, maybe we should validate only parts of the document (by XPath)?

I'm asking because I faced this problem for the third time

File: lib/asciidoctor/standoc/macros.rb (hunk 1 above)
Comment (ronaldtse) [reply lib/asciidoctor/standoc/macros.rb]: OK, but can you document this status in a code comment on top of this line?

**Comment by ronaldtse** [comment]
@camobap795 you need to update to the newest version of relaton to fetch ISO documents. Their website updated and only the latest relaton fixes it.

**Comment by CAMOBAP795** [comment]
> you need to update to the newest version of relaton to fetch ISO documents. Their website updated and only the latest relaton fixes it.

@ronaldtse  But this only short-term fix, maybe you have some idea about the long-term solution?

File: lib/asciidoctor/standoc/macros.rb (hunk 1 above)
Comment (CAMOBAP795) [reply lib/asciidoctor/standoc/macros.rb]: Done

**Comment by ronaldtse** [comment]
@abunashir is working on a longer term fix but we’ll see.

**Comment by ronaldtse** [comment]
@CAMOBAP795 can we fix the specs and merge this? Thanks.

**Comment by CAMOBAP795** [comment]
@ronaldtse I can but what's the point if it can be changed tomorrow?

I mean could you please just confirm that this need to be fixed in specs?

**Comment by ronaldtse** [comment]
@CAMOBAP795 so the specs rely on Relaton, which has recently changed.

This is the diff for one of the specs (ISO 124):

```diff
<   <title type="title-main" format="text/plain" language="en" script="Latn"></title>
<   <title type="main" format="text/plain" language="en" script="Latn"></title>
---
>   <title type="title-intro" format="text/plain" language="en" script="Latn">Latex, rubber</title>
>   <title type="title-main" format="text/plain" language="en" script="Latn">Determination of total solids content</title>
>   <title type="main" format="text/plain" language="en" script="Latn">Latex, rubber - Determination of total solids content</title>
>   <title type="title-intro" format="text/plain" language="fr" script="Latn">Latex de caoutchouc</title>
>   <title type="title-main" format="text/plain" language="fr" script="Latn">Détermination des matières solides totales</title>
>   <title type="main" format="text/plain" language="fr" script="Latn">Latex de caoutchouc - Détermination des matières solides totales</title>
6a11
>   <docidentifier type="ISO">ISO 124:2014</docidentifier>
19a25
>   <language>fr</language>
21a28
>   <abstract format="text/plain" language="fr" script="Latn">L'ISO 124:2014 spécifie des méthodes pour la détermination des matières solides totales dans le latex de plantation, le latex de concentré de caoutchouc naturel et le latex de caoutchouc synthétique. Ces méthodes ne conviennent pas nécessairement au latex d'origine naturelle autre que celui de l'Hevea brasiliensis, au latex vulcanisé, aux mélanges de latex, ou aux dispersions artificielles de caoutchouc.</abstract>
48,49c55,56
<     <structuredidentifier>
<       <project-number>?</project-number>
---
>     <structuredidentifier type="ISO">
>       <project-number>ISO 124</project-number>
```

The first part (the new parts) are correct, Relaton is now returned a lot more detail. So please adopt these changes:

```diff
<   <title type="title-main" format="text/plain" language="en" script="Latn"></title>
<   <title type="main" format="text/plain" language="en" script="Latn"></title>
---
>   <title type="title-intro" format="text/plain" language="en" script="Latn">Latex, rubber</title>
>   <title type="title-main" format="text/plain" language="en" script="Latn">Determination of total solids content</title>
>   <title type="main" format="text/plain" language="en" script="Latn">Latex, rubber - Determination of total solids content</title>
>   <title type="title-intro" format="text/plain" language="fr" script="Latn">Latex de caoutchouc</title>
>   <title type="title-main" format="text/plain" language="fr" script="Latn">Détermination des matières solides totales</title>
>   <title type="main" format="text/plain" language="fr" script="Latn">Latex de caoutchouc - Détermination des matières solides totales</title>
6a11
>   <docidentifier type="ISO">ISO 124:2014</docidentifier>
19a25
>   <language>fr</language>
21a28
>   <abstract format="text/plain" language="fr" script="Latn">L'ISO 124:2014 spécifie des méthodes pour la détermination des matières solides totales dans le latex de plantation, le latex de concentré de caoutchouc naturel et le latex de caoutchouc synthétique. Ces méthodes ne conviennent pas nécessairement au latex d'origine naturelle autre que celui de l'Hevea brasiliensis, au latex vulcanisé, aux mélanges de latex, ou aux dispersions artificielles de caoutchouc.</abstract>
```

However, this section is wrong (caused by https://github.com/metanorma/relaton-iso/issues/64), so please don't update this part.
```diff
48,49c55,56
<     <structuredidentifier>
<       <project-number>?</project-number>
---
>     <structuredidentifier type="ISO">
>       <project-number>ISO 124</project-number>
```

I'm going to merge this first and make a new issue.

**Comment by opoudjis** [comment]
I have addressed the specs issue.

I'd like to remind @ronaldtse and @CAMOBAP795 that the longer I don't have a robust release script, the longer you guys are going to be impacted by dev version breakages. And I will not switch to dev branch in ANYTHING until I have a robust release script. My life is unworkable enough with this stack as is.

**Comment by CAMOBAP795** [comment]
@opoudjis I understand this, I'm working hard to finish it ASAP

**Comment by opoudjis** [comment]
Thank you @CAMOBAP795.

**Pull request merged**